### PR TITLE
Trigger dependency updates for builds added to channels through the R…

### PR DESCRIPTION
**Pull request by riarenas**
…eleasePipelineRunner

https://github.com/dotnet/arcade/issues/2308

So far I validated locally by checking that the call to StartDependencyUpdater plumbs through to the DependencyUpdater, adds an item to its internal queue, and calls UpdateDependenciesAsync with the build/channel combination.

Haven't been able to create a subscription from my local instance, so UpdateDependenciesAsync doesn't find a subscription for it, but at that point it's all up to the dependencyUpdater to do the right thing for the BuildChannel that it's checking.

Could use some feedback on how I glued the two services together @alexperovich

I still want to see if I can modify some of the scenario tests @mmitche mentioned so we are guarded from missing a bug like this again, but it might be OK to have that as a followup??

### CODE REVIEW

File: src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs [hunk 1]
```diff
@@ -301,11 +314,12 @@ private async Task<AzureDevOpsClient> GetAzureDevOpsClientForAccount(string acco
             return new AzureDevOpsClient(accessToken, Logger, null);
         }
 
-        private void AddFinishedBuildChannelsIfNotPresent(HashSet<BuildChannel> buildChannelsToAdd)
+        private async Task<List<BuildChannel>> AddFinishedBuildChannelsIfNotPresent(HashSet<BuildChannel> buildChannelsToAdd)
         {
             var missingBuildChannels = buildChannelsToAdd.Where(x => !Context.BuildChannels.Any(y => y.ChannelId == x.ChannelId && y.BuildId == x.BuildId)).ToList();
             Context.BuildChannels.AddRange(missingBuildChannels);
```
Comment (mmitche) [inline src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs]: Should this be AddRangeAsync?

Comment (riarenas) [reply src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs]: AddRangeAsync comes with this remark and it didn't sound like it was applicable?
```
//     This method is async only to allow special value generators, such as the one
//     used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo',
//     to access the database asynchronously. For all other cases the non async method
//     should be used.
```

Comment (mmitche) [reply src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs]: I thought i saw another case where we added with AddRangeAsync (to BuildChannels).

Comment (riarenas) [reply src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs]: We have one AddRangeAsync [here](https://github.com/dotnet/arcade-services/blob/master/src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildsController.cs#L165) 

and there are a lot of AddAsync calls throughout, which also come with the same remark.

I'm not informed enough to know what the downsides of calling it async are exactly, or what exactly constitutes the cases when it should vs it shouldn't be called that way, but I can try and find out more to see if it's something we should be worrying about or if it's fine to keep as is.

**Comment by riarenas** [comment]
Created https://github.com/dotnet/arcade/issues/2319 to add a scenario test that excercises the release pipeline runner and dependency updater to prevent issues like this from slipping through the cracks.

**Pull request merged**